crates/dpapi-core/src/str.rs

```diff
@@ -26,7 +29,15 @@ pub fn from_utf16_le(data: &[u8]) -> Result<String> {
 /// *Note*: this function automatically appends a NULL-char.
 pub fn encode_utf16_le(data: &str) -> Vec<u8> {
     data.encode_utf16()
-        .chain(std::iter::once(0))
+        .chain(core::iter::once(0))
         .flat_map(|v| v.to_le_bytes())
         .collect::<Vec<_>>()
 }
```

Comment:
thought: Now that you have `str_utf16_len`, you may want to encode the utf16 string in-place using a `WriteCursor` instead.

Reply:
done